pkg/config/validation.go

```diff
@@ -401,7 +401,7 @@ func validateTLSOptions(tls *networking.Server_TLSOptions) (errs error) {
 		return
 	}
 
-	if (tls.Mode == networking.Server_TLSOptions_SIMPLE || tls.Mode == networking.Server_TLSOptions_MUTUAL) && tls.CredentialName != "" {
+	if (tls.Mode == networking.Server_TLSOptions_SIMPLE || tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL || tls.Mode == networking.Server_TLSOptions_MUTUAL) && tls.CredentialName != "" {
```

Comment:
i would suggest to break down the if condition and return early for each case.

plus, we should check when `ISTIO_MUTUAL` is used, no TLS values are set.